src/main/java/com/importer/importer/service/StudentService.java

```diff
@@ -0,0 +1,88 @@
+package com.importer.importer.service;
+
+import com.importer.importer.dto.StudentCreationDto;
+import com.importer.importer.dto.StudentDto;
+import com.importer.importer.repository.LogRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+
+import java.util.Arrays;
+import java.util.List;
+
+@Service
+public class StudentService {
+
+    @Autowired
+    private LogRepository logRepository;
+    @Transactional
+    public String postAllStudents(List<StudentCreationDto> studentCreationDtos){
+        StringBuilder responseBuilder = new StringBuilder();
+        HttpHeaders headers = new HttpHeaders();
+        Jwt jwt = (Jwt) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        String jwtToken = jwt.getTokenValue();
+        headers.add("Authorization", "Bearer "+ jwtToken );
+        headers.setAccept(List.of(MediaType.APPLICATION_JSON));
+        for(StudentCreationDto studentCreationDto : studentCreationDtos){
+
+            HttpEntity<StudentCreationDto> entity = new HttpEntity<StudentCreationDto>(studentCreationDto, headers);
+
+            String status = "";
+            String responseBody = "";
+            int statusCode = 0;
+            String responseMessage = "";
+            try {
+                ResponseEntity<String> responseEntity = new RestTemplate().exchange("http://localhost:8086/students", HttpMethod.POST, entity, String.class);
+                responseBuilder.append("Status Code: ").append(responseEntity.getStatusCode()).append(System.lineSeparator());
+                responseBuilder.append("Response Body: ").append(responseEntity.getBody()).append(System.lineSeparator());
+                status = responseEntity.getStatusCode().toString();
+                statusCode = Integer.parseInt(status.substring(0, status.indexOf(" ")));
+                responseMessage = status.substring(status.indexOf(" ")+1);
+                responseBody = "Student Created Successfully";
+
+            } catch (HttpClientErrorException.BadRequest ex) {
```

Comment:
Error handling is wrong, please correct it

Reply:
I made some changes please review it again